prov/util/src/util_domain.c

```diff
@@ -112,6 +112,8 @@ util_domain_init(struct util_domain *domain, const struct fi_info *info,
 	domain->info_domain_caps = info->caps | info->domain_attr->caps;
 	domain->info_domain_mode = info->mode | info->domain_attr->mode;
 	domain->mr_mode = info->domain_attr->mr_mode;
+	domain->tx_msg_order = info->tx_attr->msg_order;
+	domain->rx_msg_order = info->rx_attr->msg_order;
```

Comment:
I was wondering if we could OR in the msg_order values every time we allocate an endpoint.  That at least updates the tx/rx_msg_order fields.  However, it's possible for the app to register memory prior to creating any endpoints.  In that case, we would incorrectly use relaxed ordering.

So, I'm wondering if we have to have a domain that only supports relaxed ordering.  If the app calls fi_info with no ordering set, we can return that domain as one of the options.  Endpoints allocated on that domain are only unordered.

This brings up that I think most apps need ordering for sends.  AFAICT, the relax ordering here mostly impacts RMA ordering: WAW, RAR, etc.  We may only need to check that those bits are not set.